web/static/js/application.js

```diff
@@ -346,13 +346,22 @@
             $('#project_input').on('change', function () {
                 var newProject = this.value;
 
+                // Show the spinner.
+                var $loader = $('span.loader');
+                $('label[for="project_input"]').append($loader);
+                $loader.removeClass('hidden');
```

Comment:
Hm, good point. The problem with not showing it while the project is being validated is that when the validation fails they get an error, but it's not till a few seconds later sometimes, and after they've moved to the next field (at which point the project field switches back to what was there before, which is weird I think; doesn't let them correct a typo).

Maybe we could have it spin in the project field, and then do a green tick or red cross depending on what the result is? And maybe move the spinner into the text input itself (or out the other side of it)?

I'm not sure what the best approach is.